edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -191,17 +198,17 @@ public void Dispose()
             this.refreshCacheTask?.Dispose();
         }
 
-        internal Task<Option<ServiceIdentity>> GetServiceIdentityFromService(string id)
+        internal Task<Option<ServiceIdentity>> GetServiceIdentityFromService(string targetId, string onBehalfOfDevice)
```

Comment:
AFAIK we would like to stop all bad inputs in public methods.  Please make sure all callers to this method provides a valid value for targetId and onBehalfOfDevice, e.g. targetId cannot be null or empty.